activerecord/lib/active_record/associations/builder/collection_association.rb

```diff
@@ -66,17 +66,5 @@ def #{name.to_s.singularize}_ids=(ids)
         end
       CODE
     end
-
-    def self.wrap_scope(scope, mod)
```

Comment:
The fact that you removed this and nothing breaks concerns me. This seems like something we need to do otherwise the association scope will not extend the module passed in the `:extend` option. Maybe we are missing test coverage. Can you check if this would make any difference in the public API?